crates/oxc_linter/src/context.rs

```diff
@@ -29,6 +31,14 @@ impl<'a> LintContext<'a> {
         self.diagnostics.borrow_mut().push(diagnostic.into());
     }
 
+    pub fn push_fix(&mut self, fix: Fix) {
```

Comment:
Let's use `RefCell` and remove the `mut`, we may move all of these into `rayon`, it'll be easier to modify code without `mut`